corehq/apps/reports/standard/sms.py

```diff
@@ -707,6 +708,12 @@ def get_queryset(self):
                 Q(messagingsubevent__content_type__in=content_type_filter)),
         )
 
+        event_status = EventStatusFilter.get_value(self.request, self.domain)
+        if event_status is not None:
+            data = data.filter(
```

Comment:
this filter is going to have to be a little more complex - an error occurred if any of the following is true:

    status=MessagingEvent.STATUS_ERROR
    messagingsubevent__status=MessagingEvent.STATUS_ERROR
    messagingsubevent__sms__error=True

similarly, an event can be in progress if any of the following is true:

    status=MessagingEvent.STATUS_IN_PROGRESS
    messagingsubevent__status=MessagingEvent.STATUS_IN_PROGRESS
    messagingsubevent__xforms_session__end_time__isnull=True

and an event is not completed when _all_ of the following are true:

    messagingsubevent__xforms_session__end_time__isnull=False
    messagingsubevent__xforms_session__submission_id__isnull=True

and an event is completed if none of the above is the case. i think it might be easier to just allow filtering on everything, error, in progress, or not completed, and then add the right filters above conditionally for the last 3 choices